packages/components/src/utils/hooks.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Custom hooks for "controlled" components to track and consolidate internal
+ * state and incoming values. This is useful for components that render
+ * `input`, `textarea`, or `select` HTML elements.
+ *
+ * https://reactjs.org/docs/forms.html#controlled-components
+ *
+ * @param {any} initialState The initial state value.
+ * @return {Array<any, Function>} The controlled value and the value setter.
+ */
+export function useControlledState( initialState ) {
+	const [ state, setState ] = useState( initialState );
+	const stateRef = useRef( initialState );
+
+	useEffect( () => {
+		/**
+		 * Update the internal state if the incoming value changes.
+		 */
```

Comment:
At one point the JavaScript standards were updated to not be opinionated one way or the other of multi-line comments being formatted as `/*` vs. `//`, but the one which wouldn't be valid except in the case of documenting a function or variable would be the JSDoc `/**` syntax here.

```suggestion
		// Update the internal state if the incoming value changes.
```

```suggestion
		/*
		 * Update the internal state if the incoming value changes.
		 */
```